libs/vkd3d-shader/vkd3d_shader_main.c

```diff
@@ -980,3 +980,29 @@ int vkd3d_shader_parse_root_signature_v_1_2_from_raw_payload(const struct vkd3d_
     return vkd3d_shader_parse_root_signature_for_version(dxbc, out_desc, VKD3D_ROOT_SIGNATURE_VERSION_1_2, true,
             compatibility_hash);
 }
+
+vkd3d_shader_hash_t vkd3d_root_signature_v_1_2_compute_layout_compat_hash(
+        const struct vkd3d_root_signature_desc2 *desc)
+{
+    vkd3d_shader_hash_t hash = hash_fnv1_init();
+    uint32_t i;
+
+    hash = hash_fnv1_iterate_u32(hash, desc->static_sampler_count);
+    hash = hash_fnv1_iterate_u32(hash, desc->parameter_count);
+    hash = hash_fnv1_iterate_u32(hash, desc->flags & D3D12_ROOT_SIGNATURE_FLAG_LOCAL_ROOT_SIGNATURE);
+
+    for (i = 0; i < desc->parameter_count; i++)
+    {
+        hash = hash_fnv1_iterate_u32(hash, desc->parameters[i].parameter_type);
+        if (desc->parameters[i].parameter_type == VKD3D_ROOT_PARAMETER_TYPE_32BIT_CONSTANTS)
+            hash = hash_fnv1_iterate_u32(hash, desc->parameters[i].constants.value_count);
+        else
+            hash = hash_fnv1_iterate_u32(hash, 0);
+    }
+
+    /* static sampler desc are tightly packed in memory. */
+    hash = hash_fnv1_iterate_data(hash, desc->static_samplers,
+            desc->static_sampler_count * sizeof(*desc->static_samplers));
```

Comment:
`static_sampler_desc` still contains register space+number information, is it intended to include those?

Reply:
That is true ... I suppose we need more intrusive hashing.

Reply:
Fixed.